test/parallel_api/numeric/numeric.ops/reduce_by_segment_zip.pass.cpp

```diff
@@ -108,6 +108,65 @@ test_with_usm()
     EXPECT_EQ_N(exp_values1, output_values1, n, "wrong values1 from reduce_by_segment");
     EXPECT_EQ_N(exp_values2, output_values2, n, "wrong values2 from reduce_by_segment");
 }
+
+template <typename KernelName>
+void
+test_zip_with_discard()
+{
+    constexpr sycl::usm::alloc alloc_type = sycl::usm::alloc::device;
+    sycl::queue q = TestUtils::get_test_queue();
+
+    constexpr int n = 5;
+
+    //data initialization
+    int keys1[n] = {1, 1, 2, 2, 3};
+    int keys2[n] = {1, 1, 2, 2, 3};
+    int values1[n] = {1, 1, 1, 1, 1};
+    int values2[n] = {2, 2, 2, 2, 2};
+    int output_keys[n] = {};
+    int output_values1[n] = {};
+    int output_values2[n] = {};
+
+    // allocate USM memory and copying data to USM shared/device memory
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper1(q, keys1, n);
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper2(q, keys2, n);
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper3(q, values1, n);
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper4(q, values2, n);
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper5(q, output_keys, n);
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper6(q, output_values1, n);
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper7(q, output_values2, n);
+    auto d_keys1 = dt_helper1.get_data();
+    auto d_keys2 = dt_helper2.get_data();
+    auto d_values1 = dt_helper3.get_data();
+    auto d_values2 = dt_helper4.get_data();
+    auto d_output_keys = dt_helper5.get_data();
+    auto d_output_values1 = dt_helper6.get_data();
+    auto d_output_values2 = dt_helper7.get_data();
+
+    //make zip iterators
+    auto begin_keys_in = oneapi::dpl::make_zip_iterator(d_keys1, d_keys2);
+    auto end_keys_in = oneapi::dpl::make_zip_iterator(d_keys1 + n, d_keys2 + n);
+    auto begin_vals_in = oneapi::dpl::make_zip_iterator(d_values1, d_values2);
+    auto begin_keys_out = oneapi::dpl::make_zip_iterator(d_output_keys, oneapi::dpl::discard_iterator());
+    auto begin_vals_out = oneapi::dpl::make_zip_iterator(d_output_values1, d_output_values2);
```

Comment:
Does it make sense to include a discard iterator in the values as well?  I think right now we are just hitting a compilation issue just with the key side of the issue, not the value side, right?

Reply:
Yes, it makes sense to add this for completeness as the same issue would occur with a discard iterator in the output values. I have added it.